smarts/core/smarts.py

```diff
@@ -255,39 +255,40 @@ def _step(self, agent_actions):
         return observations, rewards, dones, extras
 
     def _teardown_done_agents_and_vehicles(self, dones):
-        def is_agent_done(agent_id, done):
-            if self._agent_manager.is_boid_agent(agent_id):
-                # Agent is done when all remaining vehicles are done
-                return all(done.values())
-
-            return done
-
-        def done_vehicle_ids(dones, vehicle_index):
+        def done_vehicle_ids(dones):
             vehicle_ids = set()
             for agent_id, done in dones.items():
                 if self._agent_manager.is_boid_agent(agent_id):
-                    for vehicle_id in done:
-                        if done[vehicle_id]:
-                            vehicle_ids.add(vehicle_id)
+                    vehicle_ids.update([id_ for id_ in done if done[id_]])
                 elif done:
```

Comment:
Looking at this, it seems awkward to me that these are handled so differently.

Reply:
It's due to boids having a second level dictionary (of vehicles).